x-pack/libbeat/management/managerV2.go

```diff
@@ -338,6 +346,16 @@ func (cm *BeatV2Manager) handleOutputReload(unit *client.Unit) {
 	}
 	cm.logger.Debugf("Got Output unit config '%s'", rawConfig.GetId())
 
+	// if needed, stop the beat, let agent restart
```

Comment:
We are sure this only triggers when the actual output configuration changes? And it won't restart for changes in the unit that are not the actual output configuration, like the log level? https://github.com/elastic/elastic-agent-client/blob/15881a8e64ef95f32f688c933319cd926ff500b0/elastic-agent-client.proto#L280

Reply:
Ah, didn't think of that! Should probably add some kind of check to diff the actual config.